packages/api/src/modules/providers/subventia/subventia.service.test.ts

```diff
@@ -158,37 +230,119 @@ describe("SubventiaService", () => {
         });
 
         it("should call parse", () => {
-            SubventiaService.processSubventiaData(filePath);
+            SubventiaService.processSubventiaData(filePath, exportDate);
             expect(mockParse).toHaveBeenCalledWith(filePath);
         });
 
         it("should call sortDataByValidity", () => {
-            SubventiaService.processSubventiaData(filePath);
+            SubventiaService.processSubventiaData(filePath, exportDate);
             expect(mockSortData).toHaveBeenCalledWith(parsedData);
         });
 
         it("should call getApplications", () => {
-            SubventiaService.processSubventiaData(filePath);
-            expect(mockGetApplications).toHaveBeenCalledWith(sortedData["valids"]);
+            SubventiaService.processSubventiaData(filePath, exportDate);
+            expect(mockGetApplications).toHaveBeenCalledWith(sortedData["valids"], exportDate);
         });
     });
-
     describe("createEntity", () => {
-        let mockCreate: jest.SpyInstance;
-
         const entity = {
             name: "I'm subventia entity",
         } as unknown as SubventiaDbo;
 
-        //@ts-expect-error : test private method
-        mockCreate = jest.spyOn(SubventiaRepository, "create").mockResolvedValue("FAKE_ID");
-        afterAll(() => {
-            mockCreate.mockRestore();
-        });
-
         it("should call create", async () => {
             await SubventiaService.createEntity(entity);
             expect(mockCreate).toHaveBeenCalledWith(entity);
         });
     });
+
+    /**
+     * |-------------------------|
+     * |   Demande Part          |
+     * |-------------------------|
+     */
+
+    describe("getDemandeSubventionBySiret", () => {
+        const siret = "123456789";
+        it("should call findBySiret", async () => {
+            await SubventiaService.getDemandeSubventionBySiret(siret);
+            expect(mockFindBySiret).toHaveBeenCalledWith(siret);
+        });
+
+        it("should call toDemandeSubventionDto twice", async () => {
```

Comment:
`twice` c'est surtout `for each result` non ? Ou `as many times as results`. Le fait de mettre `twice` dans le nom du test ça donne l'impression que cette valeur a du sens pour la méthode

Reply:
yep je supprimerai "twice" et utiliserait `applications.length`